components/homme/src/share/parallel_mod.F90

```diff
@@ -176,20 +170,33 @@ function initmp(npes_in,npes_stride) result(par)
     if(par%rank .eq. par%root) par%masterproc = .TRUE.
     if (par%masterproc) write(iulog,*)'number of MPI processes: ',par%nprocs
     if (par%masterproc) write(iulog,*)'MPI processors stride: ',npes_cam_stride
+#else
+    par%root          =  0
+    par%rank          =  0
+    par%nprocs        =  1
+    par%comm          = -1
+    par%masterproc    = .TRUE.
+#endif
+  end subroutine init_par
+
+  subroutine initmp_from_par(par)
+    type (parallel_t),intent(in):: par
+    character(len=MPI_MAX_PROCESSOR_NAME)               :: my_name
+    character(len=MPI_MAX_PROCESSOR_NAME), allocatable  :: the_names(:)
+    integer(kind=int_kind),allocatable                  :: tarray(:)
+    integer(kind=int_kind)                              :: namelen,i
+    integer(kind=int_kind)                              :: ierr,tmp_min,tmp_max
+    integer :: node_color
            
+#ifdef _MPI
+#include <mpif.h>
     if (MPI_DOUBLE_PRECISION==20 .and. MPI_REAL8==18) then
        ! LAM MPI defined MPI_REAL8 differently from MPI_DOUBLE_PRECISION
        ! and LAM MPI's allreduce does not accept on MPI_REAL8
        MPIreal_t    = MPI_DOUBLE_PRECISION
     else
        MPIreal_t    = MPI_REAL8
     endif
-
```

Comment:
Why were these lines deleted? Without them MPI2_real is not defined. It is used now in theta-l runs, so, I am not sure how tests got passed.

Reply:
My bad. When I split the method in two I copied and pasted, and I must have forgotten this line. I will bring it back now.